File: example/eg1/pattern_in_random.py
#!/usr/bin/python 

import sys
import random 

if len(sys.argv)<2:
    print "Usage: %s pattern " % sys.argv[0]
    print "   eg. %s 7777 " % sys.argv[0]
    sys.exit(1)

pattern=sys.argv[1]
cnt=0
random.seed()
go_on=True
while go_on:
    s=str.format("{0}",random.randrange(0,100000000))
    if s.endswith(pattern): 
        print "| Python | pattern-length:%d | iterations:%d | found:%s |" % ( len(pattern), cnt,s ) 
        go_on=False
    cnt=cnt+1

File: example/eg1/pattern_in_random.m
% by default look for 4 zeroes
pattern="0000"; % to match at the end

% arguments received from the command line
arg_list = argv();
if nargin>0
    pattern=arg_list{1};
end

cnt=0;
go_on=1;
while (go_on==1)  
    v=num2str(floor(abs(normrnd(0,10000000))));
    cnt=cnt+1;
    if (length(v)>length(pattern)) && (rindex(v,pattern)==(length(v)-length(pattern)+1))
        go_on=0;
        printf('| Octave | pattern-length:%d | iterations:%d | found:%s |', length(pattern), cnt,v ) 
    end
end

